code/modules/projectiles/gun.dm

```diff
@@ -331,83 +345,81 @@
 
 //called after successfully firing
-/obj/item/gun/proc/handle_post_fire(mob/user, atom/target, var/pointblank=0, var/reflex=0, var/obj/projectile)
+/obj/item/gun/proc/handle_post_fire(atom/movable/firer, atom/target, var/pointblank=0, var/reflex=0, var/obj/projectile)
 	if(fire_anim)
 		flick(fire_anim, src)
 
 	if(!silenced && check_fire_message_spam("fire"))
 		var/user_message = SPAN_WARNING("You fire \the [src][pointblank ? " point blank":""] at \the [target][reflex ? " by reflex" : ""]!")
 		if (silenced)
-			to_chat(user, user_message)
-		else
-			user.visible_message(
-				SPAN_DANGER("\The [user] fires \the [src][pointblank ? " point blank":""] at \the [target][reflex ? " by reflex" : ""]!"),
+			to_chat(firer, user_message)
+		else 
+			firer.visible_message(
+				SPAN_DANGER("\The [firer] fires \the [src][pointblank ? " point blank":""] at \the [target][reflex ? " by reflex" : ""]!"),
 				user_message,
 				SPAN_DANGER("You hear a [fire_sound_text]!")
 			)
 
-		if (pointblank && ismob(target))
-			admin_attack_log(user, target,
+		if (pointblank && ismob(target) && isliving(firer))
+			admin_attack_log(firer, target,
 				"shot point blank with \a [type]",
 				"shot point blank with \a [type]",
 				"shot point blank (\a [type])"
 			)
 
-	if(one_hand_penalty)
-		if(!src.is_held_twohanded(user))
-			switch(one_hand_penalty)
-				if(4 to 6)
-					if(prob(50)) //don't need to tell them every single time
-						to_chat(user, "<span class='warning'>Your aim wavers slightly.</span>")
-				if(6 to 8)
-					to_chat(user, "<span class='warning'>You have trouble keeping \the [src] on target with just one hand.</span>")
-				if(8 to INFINITY)
-					to_chat(user, "<span class='warning'>You struggle to keep \the [src] on target with just one hand!</span>")
-		else if(!user.can_wield_item(src))
-			switch(one_hand_penalty)
-				if(4 to 6)
-					if(prob(50)) //don't need to tell them every single time
-						to_chat(user, "<span class='warning'>Your aim wavers slightly.</span>")
-				if(6 to 8)
-					to_chat(user, "<span class='warning'>You have trouble holding \the [src] steady.</span>")
-				if(8 to INFINITY)
-					to_chat(user, "<span class='warning'>You struggle to hold \the [src] steady!</span>")
-
-	if(screen_shake)
-		shake_camera(user, max(burst_delay*burst, fire_delay), screen_shake)
-
-	if(combustion)
-		var/turf/curloc = get_turf(src)
-		if(curloc)
-			curloc.hotspot_expose(700, 5)
-
-	if(istype(user,/mob/living/carbon/human) && user.is_cloaked()) //shooting will disable a rig cloaking device
-		var/mob/living/carbon/human/H = user
-		if(istype(H.back,/obj/item/rig))
-			var/obj/item/rig/R = H.back
-			for(var/obj/item/rig_module/stealth_field/S in R.installed_modules)
-				S.deactivate()
-
-	if(space_recoil)
-		if(!user.check_space_footing())
-			var/old_dir = user.dir
-			user.inertia_ignore = projectile
-			step(user,get_dir(target,user))
-			user.set_dir(old_dir)
+	if(isliving(firer))
```

Comment:
It might be worth passing this entire chunk of user-specific code into a new proc for the sake of readability.